sys/kern/fdt.c

```diff
@@ -130,14 +144,58 @@ ssize_t FDT_getencprop(phandle_t node, const char *propname, pcell_t *buf,
   if (buflen % 4)
     return -1;
 
-  ssize_t ret = FDT_getprop(node, propname, buf, buflen);
-  if (ret < 0)
+  ssize_t rv = FDT_getprop(node, propname, buf, buflen);
+  if (rv < 0)
     return -1;
 
-  for (size_t i = 0; i < buflen / sizeof(uint32_t); i++)
-    buf[i] = be32toh(buf[i]);
+  FDT_decode(buf, buflen / sizeof(pcell_t));
+
+  return rv;
+}
+
+ssize_t FDT_getprop_alloc_multi(phandle_t node, const char *propname, int elsz,
+                                void **bufp) {
+  void *buf = NULL;
+  int rv = -1;
+
+  int len = FDT_getproplen(node, propname);
+  if ((len == -1) || (len % elsz))
+    goto end;
+
+  if (len) {
+    buf = kmalloc(M_DEV, len, M_WAITOK);
+    if (FDT_getprop(node, propname, buf, len) == -1) {
+      FDT_free(buf);
+      buf = NULL;
+      goto end;
+    }
+  }
+  rv = len / elsz;
+
+end:
+  *bufp = buf;
+  return rv;
```

Comment:
You can just return `-1` and leave `*bufp` unchanged in case of error. For me it will be more readable.

Reply:
I agree. It makes the code much more readable. Done.